threescale/options.go

```diff
@@ -19,9 +19,17 @@ func WithExtensions(extensions Extensions) Option {
 	}
 }
 
+// WithInstrumentationCallback allows the caller to provide an optional callback function that will
+// be called with the details of the underlying request to 3scale if present as an option
```

Comment:
It will be called, but it will use another thread - we should make that clear, either here or somewhere anyone using the client would most certainly see.